.github/workflows/test_dev.yml

```diff
@@ -63,8 +66,8 @@ jobs:
         run: |
           source env/bin/activate
           ipython kernel install --name "env" --user
-          pytest tests/ -m 'not (qpu or sim)' --cov --cov-report=xml:coverage.xml
-
+          pip install pytest-xdist
+          python -m pytest tests/ src/*/tests -m 'not (qpu or sim)' --cov --cov-report=xml:coverage.xml -n auto
```

Comment:
Can we change it to `pytest tests/ src/*/tests -m 'not (qpu or sim)'` ?

That is, no reference to python.

Also, shall we keep using coverage? We are not really using it much to be honest and it generates tons of logs which are not very insightful @vishal-ph @shahidee44

Reply:
I'm ok with removing code cov. We can keep `-n auto` for parallelizing the tests so they run slightly faster.

But I can try removing the python and see-ing if there's any issue. I'll do it for the next PR I push.

Reply:
removed `python -m` and `--cov-report`. We can add it later if we want it later.